demosplan/DemosPlanCoreBundle/Logic/Import/Statement/AbstractStatementSpreadsheetImporter.php

```diff
@@ -0,0 +1,156 @@
+use demosplan\DemosPlanCoreBundle\Logic\Document\ElementsService;
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\CurrentProcedureService;
+use demosplan\DemosPlanCoreBundle\Logic\Statement\StatementCopier;
+use demosplan\DemosPlanCoreBundle\Logic\Statement\StatementService;
+use demosplan\DemosPlanCoreBundle\Logic\User\CurrentUserService;
+use demosplan\DemosPlanCoreBundle\Logic\User\OrgaService;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use PhpOffice\PhpSpreadsheet\Worksheet\Worksheet;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webmozart\Assert\Assert;
+
+abstract class AbstractStatementSpreadsheetImporter implements StatementSpreadsheetImporterInterface
+{
+    protected readonly NotBlank $notNullConstraint;
+
+    /**
+     * @var list<Statement>
+     */
+    protected array $generatedStatements = [];
+
+    /**
+     * {@link Tag} entities that do not yet exist in the database and were created during the import.
+     *
+     * @var list<Tag>
+     */
+    protected array $generatedTags = [];
+
+    /**
+     * @var list<ImportError>
+     */
+    protected array $errors = [];
+
+    public function __construct(
+        protected readonly CurrentProcedureService $currentProcedureService,
+        protected readonly CurrentUserService $currentUser,
+        protected readonly ElementsService $elementsService,
+        protected readonly OrgaService $orgaService,
+        protected readonly StatementCopier $statementCopier,
+        protected readonly StatementService $statementService,
+        protected readonly TranslatorInterface $translator,
+        protected readonly ValidatorInterface $validator,
+    ) {
+        $this->notNullConstraint = new NotBlank(['message' => 'segment.import.error.metadata.statement.id']);
+    }
+
+    /**
+     * @return array<int, Worksheet>
+     */
+    protected function extractWorksheets(SplFileInfo $workbookFile, int $requiredWorksheets): array
+    {
+        $workbook = IOFactory::load($workbookFile->getPathname());
+
+        $worksheets = $workbook->getAllSheets();
+        Assert::greaterThanEq(count($worksheets), $requiredWorksheets, 'Expected at least %2$s worksheets, only found %s instead.');
+
+        return $worksheets;
+    }
+
+    /**
+     * @throws CopyException
+     * @throws ClusterStatementCopyNotImplementedException
+     */
+    public function createCopy(Statement $generatedOriginalStatement): Statement
+    {
+        return $this->statementCopier->copyStatementObjectWithinProcedure(
+            $generatedOriginalStatement,
+            false,
+            true,
+            false
+        );
+    }
+
+    public function getErrorsAsArray(): array
```

Comment:
missing documentation of definition of returned array

Reply:
The return is documented in the parent interface. However, I found some instances where the typehinting for the error array was actually missing.